recognition/advanced_tree_search.py

```diff
@@ -880,3 +881,148 @@ def create_config(cls, crp, extra_config, extra_post_config, **kwargs):
     def hash(cls, kwargs):
         config, post_config = cls.create_config(**kwargs)
         return super().hash({"config": config, "exe": kwargs["crp"].speech_recognizer_exe})
+
+
+class RescoreLatticeCacheJob(rasr.RasrCommand, Job):
+    def __init__(
+        self,
+        crp: rasr.CommonRasrParameters,
+        lattice_cache: tk.Path,
+        rescorer_type: str = "single-best",
+        max_hypotheses: int = 5,
+        pruning_threshold: float = 16.0,
+        history_limit: int = 0,
+        rescoring_lookahead_scale: float = 1.0,
+        rtf: float = 1.0,
+        cpu: int = 1,
+        mem: float = 4.0,
+        use_gpu: bool = False,
+        extra_config=None,
+        extra_post_config=None,
+    ):
+        self.set_vis_name("Rescore Lattice")
+
+        kwargs = locals()
+        del kwargs["self"]
+
+        (
+            self.config,
+            self.post_config,
+        ) = RescoreLatticeCacheJob.create_config(**kwargs)
+        self.input_lattice_cache = lattice_cache
+        self.exe = self.select_exe(crp.flf_tool_exe, "flf-tool")
+        self.concurrent = crp.concurrent
+        self.use_gpu = use_gpu
+
+        self.out_log_file = self.log_file_output_path("rescoring", crp, True)
+        self.out_single_lattice_caches = dict(
+            (task_id, self.output_path("lattice.cache.%d" % task_id, cached=True))
+            for task_id in range(1, crp.concurrent + 1)
+        )
```

Comment:
Nit: I'd prefer this to be 
```suggestion
        self.out_single_lattice_caches = {
            task_id: self.output_path("lattice.cache.%d" % task_id, cached=True)
            for task_id in range(1, crp.concurrent + 1)
        }
```